File: 78.Subsets.java
class Solution {
    public List<List<Integer>> subsets(int[] nums) {
        Arrays.sort(nums);
        List<List<Integer>> res = new ArrayList();
        subsetsHelper(res, new ArrayList<Integer>(), nums, 0);
        return res;
    }
    void subsetsHelper(List<List<Integer>> res, List<Integer> curr, int[]nums, int start) {
        res.add(new ArrayList<Integer>(curr));
        for (int i=start; i<nums.length; i++) {
            curr.add(nums[i]);
            subsetsHelper(res, curr, nums, i+1);
            curr.remove(curr.size()-1);
        }
    }
}

File: 78.Subsets.py
class Solution:
    def __init__(self):
        self.res = []
    def subsets(self, nums: List[int]) -> List[List[int]]:
        self.backtracking(nums, 0, [])
        return self.res
    def backtracking(self, nums, index, path):
        self.res.append(path)
        if index < len(nums):
            for i in range(index, len(nums)):
                self.backtracking(nums, i+1, path+[nums[i]])